### PR TITLE
Deprecate xproj migration

### PR DESCRIPTION
**Pull request by drewnoakes**
Fixes some parts of #4540.

With this change, when a user loads a legacy `.xproj` project they will be notified that the project is unsupported. The project will appear in Solution Explorer unloaded with "incompatible" alongside the name.

Tested on drewnoakes/metadata-extractor-dotnet@2be0b57d9cd06b570b9f923df2774e8a904ca747

image

image

---

Marked as draft to prevent merging until we have alignment with CLI release.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs [hunk 1]
```diff
@@ -1,360 +1,70 @@
...
 using Microsoft.VisualStudio.Shell.Flavor;
 using Microsoft.VisualStudio.Shell.Interop;
 
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
-
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Xproj
 {
     [Guid(ProjectType.LegacyXProj)]
     internal sealed class MigrateXprojProjectFactory : FlavoredProjectFactoryBase, IVsProjectUpgradeViaFactory, IVsProjectUpgradeViaFactory4
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs]: This is no longer "migrate"

**Comment by davkean** [comment]
I expected this to also contain deletions of:

- MigrationError.cs
- MigrationReport.cs
- THIRD-PARTY-NOTICES.md

**Comment by davkean** [comment]
I think you'll want to also remove the unused strings that this code no longer uses.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs (hunk 1 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs]: I pondered this. Will rename to `XprojProjectFactory`.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs]: case it "XProj" can ya, that case bothers me every time.

**Comment by drewnoakes** [comment]
Thanks for the review @davkean. Updated.

I found a bunch more unused resource strings and fixed them in #4896.

**Comment by drewnoakes** [comment]
Keeping it in draft until master is 16.3 to prevent accidental merging.

**Comment by drewnoakes** [comment]
Remaining task is to update docs https://docs.microsoft.com/en-us/visualstudio/porting/port-migrate-and-upgrade-visual-studio-projects?view=vs-2017, which we need to coordinate the timing of.

**Comment by davkean** [comment]
You could probably que up a PR for that too if you want to have it just ready.

**Comment by drewnoakes** [comment]
Code updated to:
- Write an error to the upgrade log linking to documentation that explains how to manually migrate from xproj to csproj (@livarcocc to review and improve text)
- Remove implementation of `IVsProjectUpgradeViaFactory`, which simplifies code a fair bit
- Remove check for `xproj` file extension as this project factory should only be called for xproj files, further simplifying code

**Pull request merged**